akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/WithLogCapturing.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2009-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.testkit.typed.internal
+
+import scala.util.control.NonFatal
+
+import akka.annotation.InternalApi
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.Outcome
+import org.scalatest.TestSuite
+import org.slf4j.LoggerFactory
+
+/**
+ * INTERNAL API
+ *
+ * Mixin this trait to a test to make log lines appear only when the test failed.
+ * Requires Logback and configuration of [[CapturingAppender]] in logback-test.xml.
+ */
+@InternalApi private[akka] trait WithLogCapturing extends BeforeAndAfterAll { self: TestSuite =>
```

Comment:
I wonder if we shouldn't provide this as testkit-public API. I think it would be useful for user tests as well as our own.

Reply:
yes, they are nice. made it so